### PR TITLE
Fixes #18: Web extension for Firefox 47+

### LINKED ISSUE #18: Make Firefox desktop add-on e10s compatible
@karlcow mentioned getting an error in Nightly.

To investigate what's causing it and make the addon multiprocess friendly.

### PR DESCRIPTION
**Pull request by dshgna**
r? @miketaylr 

Had an issue with loading the screenshot and commented on it inline

### CODE REVIEW

File: firefox-webext/background.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+var prefix = 'https://webcompat.com/?open=1&url=';
+var screenshotData = '';
+
+chrome.browserAction.onClicked.addListener(function(tab) {
+  chrome.tabs.captureVisibleTab({format: 'png'}, function(res) {
+    screenshotData = res;
+    chrome.tabs.query({currentWindow: true, active: true}, function(tab) {
+      chrome.tabs.create({ "url": prefix + encodeURIComponent(tab[0].url)}, function(tab) {
+        chrome.tabs.executeScript({
```
Comment (dshgna) [inline firefox-webext/background.js]: The screenshot attach functionality works when debugging with a breakpoint at this point, but fails normally. I tried passing `runAt:"document_end"` but it didn't solve the problem.

File: firefox-webext/manifest.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  ],
+  "icons": {
+    "32": "icon32.png",
+    "48": "icon48.png",
+    "64": "icon64.png",
+    "128": "icon128.png"
+  },
+  "browser_action": {
+    "default_icon": {
+      "19": "icon32.png",
```
Comment (dshgna) [inline firefox-webext/manifest.json]: Are all these icons necessary? I added those that were both available in the chrome and firefox addons. However, in testing Nightly seems to require "19" on desktop or will not display an icon.

Comment (miketaylr) [inline firefox-webext/manifest.json]: That's kinda funny, but thanks for figuring that out.

File: firefox-webext/background.js (hunk 1 above)
Comment (miketaylr) [inline firefox-webext/background.js]: OK, will take a closer look soon. I'm currently looking at another screenshot attach bug on webcompat.com -- something seems broken (on the server side). I think it's possibly the same bug: https://github.com/webcompat/webcompat.com/issues/985

**Comment by miketaylr** [comment]
Todo: test this once I resolve the bug at https://github.com/webcompat/webcompat.com/issues/985.

**Comment by miketaylr** [comment]
@dshgna can you test again now that webcompat/webcompat.com#985 is fixed (to see if you can actually get an issue reported)?

**Comment by dshgna** [comment]
@miketaylr Still the same, I'm afraid :(

**Comment by miketaylr** [comment]
OK, no worries. Hopefully I can take a look today, things have been busier than normal with travel, etc.

**Comment by miketaylr** [comment]
I'm gonna go ahead and merge this and we can fix bugs we identify in future issues (since merging doesn't mean shipping quite yet).

**Comment by miketaylr** [comment]
> @miketaylr Still the same, I'm afraid :(

Ah yeah, I had thought the failure was on the web server side, not the browser. I asked in the #addons IRC channel and was told it's a known bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1254003 (basically the callback fires too early). Once that's fixed, we can submit this to the AMO.

Thanks again for working on this @dshgna!

File: firefox-webext/manifest.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{
+  "manifest_version": 2,
+  "name": "webcompat.com reporter",
+  "description": "Report website compatibility issues at webcompat.com",
+  "version": "0.2",
```
Comment (miketaylr) [inline firefox-webext/manifest.json]: We should probably bump this to 0.3, sorry for not catching before.

**Pull request merged**